src/components/intro/index.tsx

```diff
@@ -90,6 +103,59 @@ export default class Intro extends Component<Props, State> {
     }
   }
 
+  @bind
+  private onBeforeInstallPromptEvent(event: BeforeInstallPromptEvent) {
+    // Don't show the mini-infobar on mobile
+    event.preventDefault();
+
+    // Save the beforeinstallprompt event so it can be called later.
+    this.setState({ deferredPrompt: event });
+
+    // Log the event.
+    ga('send', 'event', 'pwa-install', 'available');
+
+    // Make the install button visible
+    this.installButton!.style.display = 'inline-block';
+  }
+
+  @bind
+  private async onInstallClick(event: Event) {
+    // Get the deferred beforeinstallprompt event
+    const deferredPrompt = this.state.deferredPrompt;
+
+    // If there's no deferred prompt, bail.
+    if (!deferredPrompt) return;
+
+    // Set the install source as the intro install button.
+    const installSource = 'introInstallButton';
+    this.setState({ installSource });
+
+    // Show the browser install prompt
+    deferredPrompt.prompt();
+
+    // Wait for the user to accept or dismiss the install prompt
+    const {outcome} = await deferredPrompt.userChoice;
```

Comment:
I haven't seen this syntax before: `const {outcome} = await deferredPrompt.userChoice;` Any pointers to docs about it so I can understand it better and use it in the future?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring

Reply:
Sweet - thanks.